compiler/rustc_driver_impl/src/pretty.rs

```diff
@@ -18,24 +18,27 @@ pub use self::PpMode::*;
 pub use self::PpSourceMode::*;
 use crate::abort_on_err;
 
-struct NoAnn<'tcx> {
-    tcx: Option<TyCtxt<'tcx>>,
+struct AstNoAnn;
+
+impl pprust_ast::PpAnn for AstNoAnn {}
+
+struct HirNoAnn<'tcx> {
+    tcx: TyCtxt<'tcx>,
 }
 
-impl<'tcx> pprust_ast::PpAnn for NoAnn<'tcx> {}
-impl<'tcx> pprust_hir::PpAnn for NoAnn<'tcx> {
+impl<'tcx> pprust_hir::PpAnn for HirNoAnn<'tcx> {
     fn nested(&self, state: &mut pprust_hir::State<'_>, nested: pprust_hir::Nested) {
-        if let Some(tcx) = self.tcx {
-            pprust_hir::PpAnn::nested(&(&tcx.hir() as &dyn hir::intravisit::Map<'_>), state, nested)
-        }
+        pprust_hir::PpAnn::nested(
+            &(&self.tcx.hir() as &dyn hir::intravisit::Map<'_>),
```

Comment:
This is an unnecessary `&`?

Reply:
It's necessary, if I remove it I get this:
```
error[E0277]: the trait bound `dyn rustc_hir::intravisit::Map<'_>: rustc_hir_pretty::PpAnn` is not satisfied
  --> compiler/rustc_driver_impl/src/pretty.rs:32:35
   |
32 |         pprust_hir::PpAnn::nested(&self.tcx.hir() as &dyn hir::intravisit::Map<'_>, state, nested)
   |         ------------------------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `rustc_hir_pretty::PpAnn` is not implemented for `dyn rustc_hir::intravisit::Map<'_>`
   |         |
   |         required by a bound introduced by this call
   |
   = help: the trait `rustc_hir_pretty::PpAnn` is implemented for `&dyn rustc_hir::intravisit::Map<'_>`
```